shared/middleware/files_api.rb

```diff
@@ -477,9 +477,15 @@ def files_put_file(encrypted_channel_id, filename, body)
       manifest = JSON.load manifest_result[:body]
     end
 
-    # store the new file
+    # store the new file (possibly copied from an existing file)
     if params['src']
-      new_entry_json = copy_file('files', encrypted_channel_id, filename.downcase, params['src'])
+      if filename.downcase == params['src']
```

Comment:
it seems like we should downcase `params['src']` also, otherwise it seems like moving from 'Foo' to 'foo' could give the same error.